### ISSUE TITLE
Searching for the same string multiple times yields different results

### ISSUE THREAD

**Issue by WardBrian** [quarto-dev/quarto-cli#8986]
### Bug description

We recently switched the Stan documentation (https://mc-stan.org/docs/) to use Quarto. We noticed something odd about the documentation.

When I search for a term, say, `eigen`, the first time, I only get one result:
![image](https://github.com/quarto-dev/quarto-cli/assets/31640292/49f3116e-6cc8-427d-8fc8-418af997a69a)

If I click on this page, and open the search bar again, suddenly the remaining options are available. 
![image](https://github.com/quarto-dev/quarto-cli/assets/31640292/46535f35-247a-4709-afc0-b6afe8b374b1)

Some preliminary testing seems to suggest this did not happen with the version of `search.json` from before #8570

**Comment by cscheid** [comment]
Yes, this is a result of https://github.com/quarto-dev/quarto-cli/pull/8570. The behavior is unfortunate, but intentional. I'm going to leave the issue open in case we can identify a fix, but I'd rather this be the problem than https://github.com/quarto-dev/quarto-cli/issues/8567

**Comment by WardBrian** [comment]
#8567 mentioned the idea of having a 'click-to-search' option rather than searching for each successive keystroke - is that still being considered?

**Comment by WardBrian** [comment]
Making the condition added in #8570 consider the limit:
```js
  if (now - then > kFuseMaxWait && subSearchFuse === undefined && resultsRaw.length < fuseOptions.limit) {
                                                            /* new ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ */
```
seems to result in a reasonable compromise between performance and accuracy. 

The current code assumes long search terms will ultimately collapse to a small subset of pages. I think that is true for almost any real-world site. This change would just make sure that this has actually happened before restricting future keystrokes

**Comment by cscheid** [comment]
@dragonstyle do you remember if we tried the solution that @WardBrian is suggesting?

**Comment by dragonstyle** [comment]
I don't - it does strike me as potentially worth trying. The question will be whether the 'no limit' is cut down in size fast enough to overcome the penalty of the longer string...

**Comment by WardBrian** [comment]
I tried it on a local copy of the site from #8567 (which seems to have switched to Algolia in production), and it provides reasonable performance. The first word hitches for me under any configuration, but further searches are performant now and still with the above change

**Comment by WardBrian** [comment]
This issue seems to be worse in Firefox than Chromium-based browsers, from what I can tell. Perhaps JS engine speed triggers the sub-indexing at a much earlier point?

**Comment by cscheid** [comment]
> Making the condition added in #8570 consider the limit:
> 
> ```js
>   if (now - then > kFuseMaxWait && subSearchFuse === undefined && resultsRaw.length < fuseOptions.limit) {
>                                                             /* new ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ */
> ```
> 
> seems to result in a reasonable compromise between performance and accuracy.

On additional thought, I like this. It _will_ be slower in case the full searches hit the search size limit. But the moment the full search gets to a small-enough size, then we're safe to use all results.

**Comment by WardBrian** [comment]
Great! Sorry for not being up to date on the development pipeline here, but does this mean that change will be included in a 1.4.x release, or is it waiting for 1.5 at this point?

**Comment by cscheid** [comment]
It'll be out on tonight's 1.5 prerelease.
